Improve server upgrade procedures (#3867)
theforeman/foreman-documentation: guides/common/modules/proc_performing-post-upgrade-tasks.adoc

[id="performing-post-upgrade-tasks_{context}"]
= Performing post-upgrade tasks

* If you cloned the default templates in {Project} to create custom templates, verify whether the default templates changed during the upgrade.
If the default templates have changed, consider updating your custom templates to reflect these changes.

* Optional: If the default provisioning templates have been changed during the upgrade, recreate any templates cloned from the default templates.
If the custom code is executed before and/or after the provisioning process, use custom provisioning snippets to avoid recreating cloned templates.
For more information about configuring custom provisioning snippets, see {ProvisioningDocURL}Creating_Custom_Provisioning_Snippets_provisioning[Creating Custom Provisioning Snippets] in _{ProvisioningDocTitle}_.

* If your {ProjectServer} shows any discovered hosts, reboot them:
. In the {ProjectWebUI}, navigate to *Hosts* > *Discovered hosts*.
. Select all discovered hosts and reboot them.

> . Repeat the process for each organization.

Author: Applied.

[id="performing-post-upgrade-tasks_{context}"]

> = Performing post-upgrade tasks

Reviewer: I think this is good enough for now.

In my ideal world we could actually create a report somewhere that tells the user "you cloned these templates and they changed in the last update". Perhaps even create notifications to review the diff. That's a very different story than what you're doing.

> * If you use templates that you created by cloning the default templates in {Project}, verify whether the default templates were changed during the upgrade.

Reviewer 2: ```suggestion
* If you cloned the default templates in {Project} to create custom templates, verify whether the default templates changed during the upgrade.
```

> If the default templates have changed, consider updating your customized templates to match the changes.

Reviewer 2: ```suggestion
If the default templates have changed, consider updating your custom templates to reflect these changes.
```

> Note that if you make manual edits to the DNS or DHCP configuration files, you must restart {Project} services with the `{foreman-maintain} service restart` command.

Reviewer 3: Given management of DNS/DHCP is disabled, you won't have to restore the configs or restart the services.

So it's only the templates you might want to restore.

* Optional: If the default provisioning templates have been changed during the upgrade, recreate any templates cloned from the default templates.
If the custom code is executed before and/or after the provisioning process, use custom provisioning snippets to avoid recreating cloned templates.

> For more information about configuring custom provisioning snippets, see {ProvisioningDocURL}Creating_Custom_Provisioning_Snippets_provisioning[Creating Custom Provisioning Snippets] in _{ProvisioningDocTitle}_.

Reviewer 3: :+1:
